### ISSUE TITLE
User-defined colormap ignored

### ISSUE THREAD

**Issue by michele-brambilla** [sciapp/gr#139]
GR does not accept a colormap provided by the user. The file attached (gr_color.txt is actually a python script)
is a slight modification of [pygrwidgetqt5_ex](https://gr-framework.org/examples/pygrwidgetqt5_ex.html)
[gr_colors.txt](https://github.com/sciapp/gr/files/6377742/gr_colors.txt)

The first two images are the expected colormaps, but the script displays the latter (in both cases)

<img width="609" alt="expected_1" src="https://user-images.githubusercontent.com/17079797/116097811-c71ab700-a6aa-11eb-965f-f33d3dcfefcd.png">

<img width="523" alt="expected_2" src="https://user-images.githubusercontent.com/17079797/116097796-c2ee9980-a6aa-11eb-9900-f247302f9138.png">

<img width="487" alt="result" src="https://user-images.githubusercontent.com/17079797/116097816-c84be400-a6aa-11eb-9d0d-514522fafe31.png">

**Comment by FlorianRhiem** [comment]
Hey @michele-brambilla,

thank you for the bug report! There are two regions of color indices used for colormaps, a smaller one between 8 and 79 and a larger one between 1000 and 1255. `gr_setcolormapfromrgb` sets the region of 1000 to 1255, but fails to set the internal variables correctly to use this region. A proper fix for this will be in the development branch of GR shortly.

A quick workaround until you can use the fixed version is to set a colormap with an index above 100 first, as the region of indices between 1000 and 1255 will be used that way. In your example, you could for example call:

```python
gr.setcolormap(103)

## User defined colormap
gr.setcolormapfromrgb(cmap)
```

and you should get the following result:
![gks](https://user-images.githubusercontent.com/4411847/116199552-89b03b00-a737-11eb-874f-b5bf95bfff12.png)

**Comment by michele-brambilla** [comment]
@FlorianRhiem I confirm the workaround works, thanks!

**Comment by FlorianRhiem** [comment]
The fix for `gr_setcolormapfromrgb` in 490e40062b38d172cf40c44915758882f39c20c0 has been merged and `gr_volume` and `gr3_createsurfacemesh` have been fixed to use the correct color index range in 6737acfae5c1115a712599f909d70268271ed1f9, so this issue should be fixed in the develop/latest builds and in the next release.

Thank you again for reporting this!
